File: model/src/main/java/org/machinelearning/swissknife/model/timeseries/TimeSeriesAnalysisRequest.java
package org.machinelearning.swissknife.model.timeseries;

import java.util.Objects;

public class TimeSeriesAnalysisRequest {

    private final TimeSeries timeSeries;
    private final int numberOfValues;

    public TimeSeriesAnalysisRequest(TimeSeries timeSeries, int numberOfValues) {
        this.timeSeries = timeSeries;
        this.numberOfValues = numberOfValues;
    }

    public TimeSeriesAnalysisRequest() {
        this(null, 0);
    }

    public TimeSeries getTimeSeries() {
        return timeSeries;
    }

    public int getNumberOfValues() {
        return numberOfValues;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TimeSeriesAnalysisRequest that = (TimeSeriesAnalysisRequest) o;
        return numberOfValues == that.numberOfValues &&
                Objects.equals(timeSeries, that.timeSeries);
    }

    @Override
    public int hashCode() {
        return Objects.hash(timeSeries, numberOfValues);
    }

    @Override
    public String toString() {
        return "TimeSeriesAnalysisRequest{" +
                "timeSeries=" + timeSeries +
                ", numberOfValues=" + numberOfValues +
                '}';
    }
}

File: engine/time_series/model/time_series_analysis_request.py
#!/usr/bin/python3

from __future__ import annotations
from time_series.model.time_series import TimeSeries


class TimeSeriesAnalysisRequest:
  '''
  Represents the request that will be sent to the time_series_analysis_service

  Attributes
    request_id (int)  - unique id assigned to a request
    time_series (TimeSeries) - time series object that is equivalent to a csv
    number_of_values (int)   - count of values to forecast
  '''


  def __init__(self, request_id: int, time_series: TimeSeries, number_of_values: int):
    self.request_id = request_id
    self.time_series = time_series
    self.number_of_values = number_of_values


  def get_request_id(self) -> int:
    return self.request_id


  def get_time_series(self) -> TimeSeries:
    return self.time_series


  def get_number_of_values(self) -> int:
    return self.number_of_values


  def __str__(self) -> str:
    return str(self.to_json())


  def __repr__(self) -> str:
    return self.__str__()


  def __eq__(self, other) -> bool:
    return isinstance(other, TimeSeriesAnalysisRequest) and\
      self.request_id == other.request_id and\
      self.time_series == other.time_series and\
      self.number_of_values == other.number_of_values


  def to_json(self) -> dict:
    return dict(requestId = self.request_id, timeSeries=self.time_series.to_json(), numberOfValues=self.number_of_values)


  @classmethod
  def from_json(cls, data: dict) -> TimeSeriesAnalysisRequest:
    request_id = int(data['requestId'])
    time_series = TimeSeries.from_json(data['timeSeries'])
    return cls(request_id, time_series, int(data['numberOfValues']))
